src/main/java/org/elasticsearch/index/query/BaseTermQueryBuilder.java

```diff
@@ -0,0 +1,242 @@
+    protected String queryName;
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, String value) {
+        this(fieldName, (Object) value);
+    }
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, int value) {
+        this(fieldName, (Object) value);
+    }
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, long value) {
+        this(fieldName, (Object) value);
+    }
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, float value) {
+        this(fieldName, (Object) value);
+    }
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, double value) {
+        this(fieldName, (Object) value);
+    }
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, boolean value) {
+        this(fieldName, (Object) value);
+    }
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, Object value) {
+        this.fieldName = fieldName;
+        if (value instanceof String) {
+            this.value = BytesRefs.toBytesRef(value);
+        } else {
+            this.value = value;
+        }
+    }
+
+    BaseTermQueryBuilder() {
+        // for serialization only
+    }
+
+    /** Returns the field name used in this query @see org.apache.lucene.search.spans.SpanTermQuery#getField() */
+    public String fieldName() {
+        return this.fieldName;
+    }
+
+    /** Returns the value used in this query @see org.apache.lucene.index.Term#text() */
+    public Object value() {
+        return this.value;
+    }
+    
+    /** Returns the query name for the query. */
+    public String queryName() {
+        return this.queryName;
+    }
+    /**
+     * Sets the query name for the query.
+     */
+    @SuppressWarnings("unchecked")
+    public QB queryName(String queryName) {
+        this.queryName = queryName;
+        return (QB) this;
+    }
+    
+    /** Returns the boost for this query. @see org.apache.lucene.search.Query#setBoost() */
```

Comment:
here too about the @see I am not sure we should refer to lucene classes here, this should be decoupled from lucene?